tests/functional/object/mcg/ui/test_bucket_policy_ui.py

```diff
@@ -0,0 +1,232 @@
+                else:
+                    pytest.skip("No OBC bucket found for account-dependent policy test")
+        else:
+            # For policies that don't require account ID, use any existing bucket
+            buckets = bucket_ui.get_buckets_list()
+            if not buckets:
+                pytest.skip("No buckets available for testing")
+
+            target_bucket_name = buckets[0]
+
+        bucket_permissions_ui = BucketsTabPermissions()
+
+        policy_method = getattr(bucket_permissions_ui, method_name)
+        policy_method(bucket_name=target_bucket_name, **params)
+
+        logger.info(f"Successfully completed {policy_name} bucket policy test")
+
+    @polarion_id("OCS-6893")
+    def test_delete_bucket_policy_ui(self, setup_ui_class_factory):
+        """
+        Test deleting bucket policy via UI.
+
+        This test uses the unified delete workflow that:
+        1. Navigates to Object Storage and bucket permissions
+        2. Verifies a policy exists before attempting deletion
+        3. Activates policy editor and deletes the policy
+        4. Handles confirmation dialog automatically
+
+        Note: This test assumes a policy already exists from previous tests.
+
+        Raises:
+            pytest.skip: If no buckets are available for testing
+        """
+        setup_ui_class_factory()
+        logger.info("Starting test to delete bucket policy")
+
+        bucket_ui = BucketsTab()
+        bucket_ui.navigate_buckets_page()
+
+        buckets = bucket_ui.get_buckets_list()
+        if not buckets:
+            pytest.skip("No buckets available for testing")
+
+        bucket_permissions_ui = BucketsTabPermissions()
+
+        # Use unified delete workflow
+        bucket_permissions_ui.delete_bucket_policy_ui(bucket_name=None)
+
+        logger.info("Successfully completed delete bucket policy test")
+
+    @polarion_id("OCS-6894")
+    @pytest.mark.parametrize("policy_name", ["AllowPublicReadAccess"])
+    def test_bucket_policy_workflow_steps(self, setup_ui_class_factory, policy_name):
+        """
+        Test individual steps of bucket policy workflow.
+
+        This test breaks down the workflow into individual steps
+        to verify each component works correctly.
+
+        Args:
+            policy_name (str): Name of the policy to test
+
+        Raises:
+            pytest.skip: If no buckets are available for testing
+        """
+        setup_ui_class_factory()
+        logger.info(f"Starting step-by-step test for policy: {policy_name}")
+
+        bucket_ui = BucketsTab()
+        bucket_ui.navigate_buckets_page()
+
+        buckets = bucket_ui.get_buckets_list()
+        if not buckets:
+            pytest.skip("No buckets available for testing")
+
+        bucket_permissions_ui = BucketsTabPermissions()
+
+        # Step 1: Navigate to bucket permissions
+        bucket_permissions_ui.navigate_to_bucket_permissions(bucket_name=None)
```

Comment:
Same comment as before - it's better to use a fresh bucket and not first.bucket. 

However, if you are using first.bucket then please add a final step to this test where you delete the bucket policy from it and restore it to the original state.